x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_factory.tsx

```diff
@@ -0,0 +1,386 @@
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const [coreStart, pluginStart] = await getStartServices();
+
+      const { http, uiSettings, notifications, ...startServices } = coreStart;
+      const { lens, data, usageCollection, fieldFormats } = pluginStart;
+
+      const deps = {
+        http,
+        uiSettings,
+        data,
+        notifications,
+        lens,
+        usageCollection,
+        fieldFormats,
+        ...startServices,
+      };
+      const {
+        api: timeRangeApi,
+        comparators: timeRangeComparators,
+        serialize: serializeTimeRange,
+      } = initializeTimeRange(state);
+
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+
+      const {
+        fieldStatsControlsApi,
+        dataLoadingApi,
+        fieldStatsControlsComparators,
+        serializeFieldStatsChartState,
+        onFieldStatsTableDestroy,
+        resetData$,
+      } = initializeFieldStatsControls(state);
+      const { onError } = dataLoadingApi;
+
+      const defaultDataViewId = await deps.data.dataViews.getDefaultId();
+      const validDataViewId: string =
+        isDefined(state.dataViewId) && state.dataViewId !== ''
+          ? state.dataViewId
+          : defaultDataViewId ?? '';
+      let initialDataView: DataView[] | undefined;
+      try {
+        const dataView = isESQLQuery(state.query)
+          ? await getESQLAdHocDataview(
+              getIndexPatternFromESQLQuery(state.query.esql),
+              deps.data.dataViews
+            )
+          : await deps.data.dataViews.get(validDataViewId);
+        initialDataView = [dataView];
+      } catch (error) {
+        // Only need to publish blocking error if viewtype is data view, and no data view found
+        if (state.viewType === FieldStatsInitializerViewType.DATA_VIEW) {
+          onError(error);
+        }
+      }
+
+      const dataViews$ = new BehaviorSubject<DataView[] | undefined>(initialDataView);
+
+      const subscriptions = new Subscription();
+      if (fieldStatsControlsApi.dataViewId$) {
+        subscriptions.add(
+          fieldStatsControlsApi.dataViewId$
+            .pipe(
+              skip(1),
+              skipWhile((dataViewId) => !dataViewId && !defaultDataViewId),
+              switchMap((dataViewId) => {
+                try {
+                  return deps.data.dataViews.get(dataViewId ?? defaultDataViewId);
+                } catch (error) {
+                  return of(undefined);
+                }
+              })
+            )
+            .subscribe((nextSelectedDataView) => {
+              if (nextSelectedDataView) {
+                dataViews$.next([nextSelectedDataView]);
+              }
+            })
+        );
+      }
+
+      const { toasts } = deps.notifications;
+
+      const api = buildApi(
+        {
+          ...timeRangeApi,
+          ...titlesApi,
+          ...fieldStatsControlsApi,
+          ...dataLoadingApi,
```

Comment:
Couple questions about exposing `dataLoadingApi` in your API. The type `FieldStatisticsTableEmbeddableApi` does not include keys for `onRenderComplete`, `onLoading`, and `onError`. Is there a reason why these keys are getting exposed via the embeddable API? Do consumers of the api use them?

Another point `dataLoading$` needs to be exposed as `dataLoading`. There is some inconsistencies in naming that we need to fix on the interface side.

Reply:
@nreese Updated here https://github.com/elastic/kibana/pull/184030/commits/2e106ba25c60fff45a853a9b50adbb7e5376f989